drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -153,12 +166,18 @@ public void init(ScenarioSimulationEditorWrapper scenarioSimulationEditorWrapper
         testRunnerReportingPanel.reset();
     }
 
+    public void setFocusedContext(ScenarioSimulationContext scenarioSimulationContext) {
+        this.focusedContext = scenarioSimulationContext;
+        populateRightDocks(TestToolsPresenter.IDENTIFIER);
```

Comment:
@danielezonca This is required to assign the current focused context to the "TestTools" dock. Please refer to `protected void setTestTools(TestToolsView.Presenter presenter)` method in the same class. This must be called anytime the focus change. Anyway, this will most probably change again with the fixing History PR, our idea with @gitgabrio is to have only a context, which will be able to manage both grid.